feat(forms): Prevents marking fields as touched/dirty when state is hidden/r…

SkyZeroZx opened the pull request:
feat(forms): prevent marking non-interactive fields as touched or dirty

##  Summary

This PR addresses the TODO comments in `FieldNodeState` class by implementing logic to prevent fields that are readonly, disabled, or hidden from being marked as touched or dirty when `markAsTouched()` or `markAsDirty()` are called.


Question : 

Should `isNonInteractive` be exposed as a public API?
Currently, `isNonInteractive` is implemented as a private computed property. 
However, there might be value in exposing this as part of the public API , benefits: Simplifies developer checks, supports conditional UI logic, streamlines form validation, and improves accessibility.

kirjs reviewed (changes requested):
Hey Jaime,

Thanks a lot for sending a PR to signal forms, i know it's not super easy while things are still in the air. 

I think this is a great start, but there are couple things to keep in mind:
1.  Imagine a case where a user maked an active field as dirty, and then made it inactive. With current implementation it would stay as dirty, while we want it to switch to pristine.
2. I'm also a bit concerned about quietly swallowing calls to markAs*, as it could lead to painful debugging sessions. 

Instead I we could use `isNonInteractive` in computing the states, and always allow the writes. 

Wdyt?

JeanMeche commented:
This probably needs rebase, we fixed the devtools regression already.

SkyZeroZx committed:
feat(forms): Prevents marking fields as touched/dirty when state is hidden/readonly/disabled

Ensures fields that are hidden, disabled, or readonly cannot be marked as touched or dirty, improving form state integrity

SkyZeroZx commented:
Updated, it would be possible to run the tests again.
